File: Easy/1154_Day_of_the_year.cpp
///first attempt 100%faster, 100%less memory
//Idea:Pay attetion to leap year
class Solution {
public:
    int dayOfYear(string date) {
    vector<int>intdate;
    int first=0;
    intdate.push_back(stoi(date.substr(0,4)));//convert string into number
    intdate.push_back(stoi(date.substr(5,2)));
    intdate.push_back(stoi(date.substr(8,2)));
    int sum=0;
    int montharray[12]={31,28,31,30,31,30,31,31,30,31,30,31};
    for(int i=0;i<intdate[1]-1;i++)
    {
        sum=sum+montharray[i];
    }
    sum=sum+intdate[2];//all the days before
    if(intdate[1]>2&&((intdate[0]%400==0)||(intdate[0]%4==0&&intdate[0]%100!=0)))//leap year
    sum++;
    return sum;

    }
};

File: Easy_python/1154_day_of_the_year.py
class Solution:
    def dayOfYear(self, date: str) -> int:
        year=int(date[:4])
        month=int(date[5:7])
        day=int(date[8:10])
       
        arrsum=0
        montharray=[31,28,31,30,31,30,31,31,30,31,30,31]
        for i in range(month-1):
            arrsum=arrsum+montharray[i]

        arrsum=arrsum+day
        if month>2 and ((year%400==0) or (year%4==0 and year%100!=0)):
            arrsum=arrsum+1
        return arrsum
